clang/test/SemaCXX/lambda-expressions.cpp

```diff
@@ -92,14 +94,14 @@ namespace ImplicitCapture {
     [] { return ref_i; }; // expected-error {{variable 'ref_i' cannot be implicitly captured in a lambda with no capture-default specified}} expected-note {{lambda expression begins here}} expected-note 2 {{capture 'ref_i' by}} expected-note 2 {{default capture by}}
 
     static int j;
-    int &ref_j = j;
-    [] { return ref_j; }; // ok
+    int &ref_j = j; // cxx03-fixme-note {{declared here}}
+    [] { return ref_j; }; // cxx03-fixme-error {{variable 'ref_j' cannot be implicitly captured in a lambda with no capture-default specified}} cxx03-fixme-note 4 {{capture}} cxx03-fixme-note {{lambda expression begins here}}
```

Comment:
I don't really understand this failure. It would be great if someone could point me at the logic for allowing the capture of `ref_j`.

Reply:
`ref_j` is not odr-used because it is a reference whose initializer is itself a
usable in constant expressions (j is static and default initialized).

Because `ref_j` is not odr-used, it does not need to be captured.

https://eel.is/c++draft/expr.const#3
https://eel.is/c++draft/expr.const#4.3

Reply:
Ah OK. So this diagnostic is actually correct, since an `int&` is not usable in a constant expression in C++03. Or am I misunderstanding more?